src/core/Object3D.js

```diff
@@ -687,7 +687,9 @@ class Object3D extends EventDispatcher {
 		if ( this.visible === false ) object.visible = false;
 		if ( this.frustumCulled === false ) object.frustumCulled = false;
 		if ( this.renderOrder !== 0 ) object.renderOrder = this.renderOrder;
-		if ( JSON.stringify( this.userData ) !== '{}' ) object.userData = this.userData;
+		
+		// eslint-disable-next-line no-unused-vars
+		for ( const key in this.userData ) { object.userData = this.userData; break };
```

Comment:
I think your first approach with `Object.keys()` was more readable and clean. I don't think there will be noticeable performance issues with objects with many properties.

Reply:
By the way, do we really want to write this with one line? I think the following may be ok because. For me it may be more readable.

```javascript
for ( const key in this.userData ) {
    object.userData = this.userData;
    break;
}
```

Reply:
We use `Object.keys()` in other `toJSON()` methods, too:

https://github.com/mrdoob/three.js/blob/42c59b4d516fd204f5f760e1f7daefbbbf18c49c/src/core/BufferGeometry.js#L871
https://github.com/mrdoob/three.js/blob/42c59b4d516fd204f5f760e1f7daefbbbf18c49c/src/materials/ShaderMaterial.js#L161

No one has complained so far about performance and I doubt somebody will.

Reply:
Anyway, I don't want to block this PR if others favor the `for` loop. Maybe it's best if @mrdoob makes a decision. 

I just think it's important to have a unify coding style. If the `for` loop is introduced, the above code sections in `BufferGeometry` and `ShaderMaterial` should be updated, too.

Reply:
Understood